scheduler/reconcile.go

```diff
@@ -1201,6 +1201,10 @@ func (a *allocReconciler) computeReconnecting(reconnecting allocSet) {
 			continue
 		}
 
+		// Record the new ClientStatus to indicate to future evals that the
+		// alloc has already reconnected.
+		alloc.AppendState(structs.AllocStateFieldClientStatus, alloc.ClientStatus)
```

Comment:
Is this alloc a new alloc to be applied via a plan? Just making sure we're not mutating an object that's in the statestore.

Reply:
Nice catch! I don't think it's being copied. Tracing it back, this `alloc` comes from these [`reconnecting[alloc.ID] = alloc`](https://github.com/hashicorp/nomad/blob/v1.4.2/scheduler/reconcile_util.go#L249). 

It's so easy to break stuff 😅